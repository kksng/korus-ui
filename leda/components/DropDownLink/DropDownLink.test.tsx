// @ts-nocheck
import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { DropDownLink } from './index';

describe('DropDownLink SNAPSHOTS', () => {
  it('should render with data objects', () => {
    const data = [
      {
        order: {
          sortField: 'createDate',
          sortOrder: 'DESC',
        },
        name: 'По умолчанию',
      },
      {
        order: {
          sortField: 'cost.commonPrice',
          sortOrder: 'ASC',
        },
        name: 'Цена по возрастанию',
      },
      {
        order: {
          sortField: 'cost.commonPrice',
          sortOrder: 'DESC',
        },
        name: 'Цена по убыванию',
      },
      {
        order: {
          sortField: 'commonArea',
          sortOrder: 'ASC',
        },
        name: 'Площадь по возрастанию',
      },
      {
        order: {
          sortField: 'commonArea',
          sortOrder: 'DESC',
        },
        name: 'Площадь по убыванию',
      },
    ];

    const valueControll = {
      order: {
        sortField: 'commonArea',
        sortOrder: 'DESC',
      },
      name: 'Площадь по убыванию',
    };

    const wrapper = mount(
      <DropDownLink
        data={data}
        onChange={(val) => val}
        value={valueControll}
        textField="name"
      />,
    );

    expect(wrapper.find('A').first().text()).toEqual('Площадь по убыванию');

    expect(wrapper.find('DropDownLink Item')).toHaveLength(5);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with array string', () => {
    const data = [
      'По умолчанию',
      'Цена по возрастанию',
      'Цена по убыванию',
      'Площадь по возрастанию',
      'Площадь по убыванию',
    ];

    const wrapper = mount(
      <DropDownLink
        data={data}
        onChange={(val) => val}
        value="По умолчанию"
      />,
    );

    expect(wrapper.find('DropDownLink Item')).toHaveLength(5);

    expect(wrapper.find('A a').last().text()).toEqual('Площадь по убыванию');

    expect(wrapper).toMatchSnapshot();
  });
});


describe('DropDownLink ATTRIBUTES', () => {
  it('should customize render items', () => {
    const data = [
      'По умолчанию',
      'Цена по возрастанию',
      'Цена по убыванию',
      'Площадь по возрастанию',
      'Площадь по убыванию',
    ];

    const wrapper = mount(
      <DropDownLink
        data={data}
        onChange={(val) => val}
        value="По умолчанию"
        itemRender={({ componentProps: { item } }) => <span>{`---${item}`}</span>}
      />,
    );

    expect(wrapper.find('DropDownLink Item').first().find('span').text()).toEqual('---По умолчанию');

    expect(wrapper.find('DropDownLink Item').last().find('span').text()).toEqual('---Площадь по убыванию');

    expect(wrapper).toMatchSnapshot();
  });

  it('should customize title', () => {
    const data = [
      'По умолчанию',
      'Цена по возрастанию',
      'Цена по убыванию',
      'Площадь по возрастанию',
      'Площадь по убыванию',
    ];

    const valueControll = 'По умолчанию';

    const wrapper = mount(
      <DropDownLink
        data={data}
        onChange={(val) => val}
        value={valueControll}
        titleRender={({ elementProps: { children } }) => <span>{`$$${children}$$`}</span>}
      />,
    );

    expect(wrapper.find('Title').text()).toEqual('$$По умолчанию$$');

    expect(wrapper).toMatchSnapshot();
  });
});

describe('ButtonGroup forward ref', () => {
  it('should return ref', () => {
    const myRef = React.createRef();

    const data = [
      'По умолчанию',
      'Цена по возрастанию',
      'Цена по убыванию',
      'Площадь по возрастанию',
      'Площадь по убыванию',
    ];

    let valueControll = 'По умолчанию';

    const handleClick = (val) => {
      valueControll = val.target.value;
    };

    mount(
      <>
        <DropDownLink
          ref={myRef}
          data={data}
          onChange={handleClick}
          value={valueControll}
          titleRender={({ elementProps: { children } }) => <span>{`$$${children}$$`}</span>}
          name="Eltsen"
        />,
      </>,
    );

    expect(myRef.current).toBeDefined();

    expect(myRef.current.wrapper.textContent).toEqual('$$По умолчанию$$По умолчаниюЦена по возрастаниюЦена по убываниюПлощадь по возрастаниюПлощадь по убыванию');
  });
});
