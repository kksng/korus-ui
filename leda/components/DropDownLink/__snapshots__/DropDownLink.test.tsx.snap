// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownLink ATTRIBUTES should customize render items 1`] = `
<DropDownLink
  data={
    Array [
      "По умолчанию",
      "Цена по возрастанию",
      "Цена по убыванию",
      "Площадь по возрастанию",
      "Площадь по убыванию",
    ]
  }
  itemRender={[Function]}
  onChange={[Function]}
  value="По умолчанию"
>
  <DropDown
    isOpen={false}
    onMouseEnter={[Function]}
    theme={
      Object {
        "wrapper": "dropdown-wrapper blank button-wrapper",
      }
    }
  >
    <Wrapper
      className="dropdown-wrapper blank button-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <Span
        className="dropdown-wrapper blank button-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <span
          className="dropdown-wrapper blank button-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <Title
            className="dropdown-text alt"
            key=".0"
          >
            <A
              className="dropdown-text alt"
            >
              <a
                className="dropdown-text alt"
                onClick={[Function]}
              >
                По умолчанию
              </a>
            </A>
          </Title>
          <Ul
            default={true}
            key=".1"
          >
            <ul
              default={true}
            >
              <Component
                className="dropdown-item level-2"
                item="По умолчанию"
                itemRender={[Function]}
                key="undefined_По умолчанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <span>
                    ---По умолчанию
                  </span>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по возрастанию"
                itemRender={[Function]}
                key="undefined_Цена по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <span>
                    ---Цена по возрастанию
                  </span>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по убыванию"
                itemRender={[Function]}
                key="undefined_Цена по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <span>
                    ---Цена по убыванию
                  </span>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по возрастанию"
                itemRender={[Function]}
                key="undefined_Площадь по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <span>
                    ---Площадь по возрастанию
                  </span>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по убыванию"
                itemRender={[Function]}
                key="undefined_Площадь по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <span>
                    ---Площадь по убыванию
                  </span>
                </Item>
              </Component>
            </ul>
          </Ul>
        </span>
      </Span>
    </Wrapper>
  </DropDown>
</DropDownLink>
`;

exports[`DropDownLink ATTRIBUTES should customize title 1`] = `
<DropDownLink
  data={
    Array [
      "По умолчанию",
      "Цена по возрастанию",
      "Цена по убыванию",
      "Площадь по возрастанию",
      "Площадь по убыванию",
    ]
  }
  onChange={[Function]}
  titleRender={[Function]}
  value="По умолчанию"
>
  <DropDown
    isOpen={false}
    onMouseEnter={[Function]}
    theme={
      Object {
        "wrapper": "dropdown-wrapper blank button-wrapper",
      }
    }
  >
    <Wrapper
      className="dropdown-wrapper blank button-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <Span
        className="dropdown-wrapper blank button-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <span
          className="dropdown-wrapper blank button-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <Title
            className="dropdown-text alt"
            key=".0"
          >
            <span>
              $$По умолчанию$$
            </span>
          </Title>
          <Ul
            default={true}
            key=".1"
          >
            <ul
              default={true}
            >
              <Component
                className="dropdown-item level-2"
                item="По умолчанию"
                key="undefined_По умолчанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          По умолчанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по возрастанию"
                key="undefined_Цена по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по убыванию"
                key="undefined_Цена по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по возрастанию"
                key="undefined_Площадь по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по убыванию"
                key="undefined_Площадь по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
            </ul>
          </Ul>
        </span>
      </Span>
    </Wrapper>
  </DropDown>
</DropDownLink>
`;

exports[`DropDownLink SNAPSHOTS should render with array string 1`] = `
<DropDownLink
  data={
    Array [
      "По умолчанию",
      "Цена по возрастанию",
      "Цена по убыванию",
      "Площадь по возрастанию",
      "Площадь по убыванию",
    ]
  }
  onChange={[Function]}
  value="По умолчанию"
>
  <DropDown
    isOpen={false}
    onMouseEnter={[Function]}
    theme={
      Object {
        "wrapper": "dropdown-wrapper blank button-wrapper",
      }
    }
  >
    <Wrapper
      className="dropdown-wrapper blank button-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <Span
        className="dropdown-wrapper blank button-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <span
          className="dropdown-wrapper blank button-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <Title
            className="dropdown-text alt"
            key=".0"
          >
            <A
              className="dropdown-text alt"
            >
              <a
                className="dropdown-text alt"
                onClick={[Function]}
              >
                По умолчанию
              </a>
            </A>
          </Title>
          <Ul
            default={true}
            key=".1"
          >
            <ul
              default={true}
            >
              <Component
                className="dropdown-item level-2"
                item="По умолчанию"
                key="undefined_По умолчанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          По умолчанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по возрастанию"
                key="undefined_Цена по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Цена по убыванию"
                key="undefined_Цена по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по возрастанию"
                key="undefined_Площадь по возрастанию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item="Площадь по убыванию"
                key="undefined_Площадь по убыванию"
                onClick={[Function]}
                textField=""
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
            </ul>
          </Ul>
        </span>
      </Span>
    </Wrapper>
  </DropDown>
</DropDownLink>
`;

exports[`DropDownLink SNAPSHOTS should render with data objects 1`] = `
<DropDownLink
  data={
    Array [
      Object {
        "name": "По умолчанию",
        "order": Object {
          "sortField": "createDate",
          "sortOrder": "DESC",
        },
      },
      Object {
        "name": "Цена по возрастанию",
        "order": Object {
          "sortField": "cost.commonPrice",
          "sortOrder": "ASC",
        },
      },
      Object {
        "name": "Цена по убыванию",
        "order": Object {
          "sortField": "cost.commonPrice",
          "sortOrder": "DESC",
        },
      },
      Object {
        "name": "Площадь по возрастанию",
        "order": Object {
          "sortField": "commonArea",
          "sortOrder": "ASC",
        },
      },
      Object {
        "name": "Площадь по убыванию",
        "order": Object {
          "sortField": "commonArea",
          "sortOrder": "DESC",
        },
      },
    ]
  }
  onChange={[Function]}
  textField="name"
  value={
    Object {
      "name": "Площадь по убыванию",
      "order": Object {
        "sortField": "commonArea",
        "sortOrder": "DESC",
      },
    }
  }
>
  <DropDown
    isOpen={false}
    onMouseEnter={[Function]}
    theme={
      Object {
        "wrapper": "dropdown-wrapper blank button-wrapper",
      }
    }
  >
    <Wrapper
      className="dropdown-wrapper blank button-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <Span
        className="dropdown-wrapper blank button-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <span
          className="dropdown-wrapper blank button-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <Title
            className="dropdown-text alt"
            key=".0"
          >
            <A
              className="dropdown-text alt"
            >
              <a
                className="dropdown-text alt"
                onClick={[Function]}
              >
                Площадь по убыванию
              </a>
            </A>
          </Title>
          <Ul
            default={true}
            key=".1"
          >
            <ul
              default={true}
            >
              <Component
                className="dropdown-item level-2"
                item={
                  Object {
                    "name": "По умолчанию",
                    "order": Object {
                      "sortField": "createDate",
                      "sortOrder": "DESC",
                    },
                  }
                }
                key="undefined_По умолчанию"
                onClick={[Function]}
                textField="name"
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          По умолчанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item={
                  Object {
                    "name": "Цена по возрастанию",
                    "order": Object {
                      "sortField": "cost.commonPrice",
                      "sortOrder": "ASC",
                    },
                  }
                }
                key="undefined_Цена по возрастанию"
                onClick={[Function]}
                textField="name"
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item={
                  Object {
                    "name": "Цена по убыванию",
                    "order": Object {
                      "sortField": "cost.commonPrice",
                      "sortOrder": "DESC",
                    },
                  }
                }
                key="undefined_Цена по убыванию"
                onClick={[Function]}
                textField="name"
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Цена по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item={
                  Object {
                    "name": "Площадь по возрастанию",
                    "order": Object {
                      "sortField": "commonArea",
                      "sortOrder": "ASC",
                    },
                  }
                }
                key="undefined_Площадь по возрастанию"
                onClick={[Function]}
                textField="name"
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по возрастанию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
              <Component
                className="dropdown-item level-2"
                item={
                  Object {
                    "name": "Площадь по убыванию",
                    "order": Object {
                      "sortField": "commonArea",
                      "sortOrder": "DESC",
                    },
                  }
                }
                key="undefined_Площадь по убыванию"
                onClick={[Function]}
                textField="name"
              >
                <Item
                  className="dropdown-item level-2"
                  onClick={[Function]}
                >
                  <Li
                    className="dropdown-item level-2"
                    onClick={[Function]}
                  >
                    <li
                      className="dropdown-item level-2"
                      onClick={[Function]}
                    >
                      <A
                        href="#"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                        >
                          Площадь по убыванию
                        </a>
                      </A>
                    </li>
                  </Li>
                </Item>
              </Component>
            </ul>
          </Ul>
        </span>
      </Span>
    </Wrapper>
  </DropDown>
</DropDownLink>
`;
